### PR TITLE
Task Card Component update

### LINKED ISSUE #429: Status Card Component
# Status card
## Overview

In Real Dev Squad, we use the [status site](https://status.realdevsquad.com/) to keep track of ongoing tasks. This a custom project tracking solution for our usecases.

It's time to refresh our designs and start making things a bit more consistent with the overall brand. So, we are starting off with updating or creating our components.

We want to implement this new design for cards that display each task.

## Design
[Figma Link](https://www.figma.com/file/NQco0iElImpT2ncuczuh2j/Status-Card?node-id=0%3A1&t=LrL6wSrQJYhw8Ypv-1)

## User stories

- As a viewer, the card should match the design provided.
- As a viewer, the card should correctly render the information to match the current implementation.
- As a superuser, the "edit" controls should be available as currently available.

## Requirements

- TDD should be followed.
- Put the component usage behind a FF(feature flag).

## Acceptance Criteria

- [ ] The new component can seemlessly be swapped out without affecting the functionality of the existing page.
- [ ] Tests coverage is exhaustive enough for the component to guarantee correctness and stability.

### PR DESCRIPTION
**Pull request by bhtibrewal**
closes #429 

[Figma Link](https://www.figma.com/file/NQco0iElImpT2ncuczuh2j/Status-Card?node-id=0%3A1&t=LrL6wSrQJYhw8Ypv-1)
Updating Task Card Component

**Before**
image

**After**
image

-------
**Before**
image

**After**
image

### Video
https://user-images.githubusercontent.com/42600164/229362697-92e5c66f-16fa-4500-b0a8-9c497156717b.mov

### CODE REVIEW

**Review by Maheima — CHANGES_REQUESTED** [review]

**Comment by Maheima** [comment]
The branch has some merge conflicts. Please resolve them

File: .env.development [hunk 1]
```diff
@@ -1,4 +1,4 @@
-NEXT_PUBLIC_BASE_URL='https://staging-api.realdevsquad.com'
+NEXT_PUBLIC_BASE_URL='https://api.realdevsquad.com'
 NEXT_PUBLIC_GITHUB_IMAGE_URL='https://raw.githubusercontent.com/Real-Dev-Squad/website-static/main/members'
```
[the remark this answers is not in the document]
Comment (bhtibrewal) [reply .env.development:2]: resolved

File: src/context/isUserAuthorized.tsx [hunk 2]
```diff
@@ -29,15 +29,15 @@ const IsUserAuthorizedContext: FC<Props> = ({ children }) => {
...
   }, []);
-  return(
+  return (
     <isUserAuthorizedContext.Provider value={isUserAuthorized}>
       {children}
     </isUserAuthorizedContext.Provider>
-  )
-}
+  );
+};
```
Comment (Maheima) [inline src/context/isUserAuthorized.tsx:41]: why is this file changed?

File: src/hooks/useKeyLongPressed.ts [hunk 3]
```diff
@@ -1,29 +1,29 @@
...
 
-    document.addEventListener('keydown', handleKeyDown);
-    document.addEventListener('keyup', handleKeyUp);
+    document.addEventListener("keydown", handleKeyDown);
+    document.addEventListener("keyup", handleKeyUp);
     return () => {
       clearTimeout(timerId);
-      document.removeEventListener('keydown', handleKeyDown);
-      document.removeEventListener('keyup', handleKeyUp);
+      document.removeEventListener("keydown", handleKeyDown);
```
Comment (Maheima) [inline src/hooks/useKeyLongPressed.ts:23]: why is this file changed?

File: src/context/isUserAuthorized.tsx (hunk 2 above)
Comment (bhtibrewal) [reply src/context/isUserAuthorized.tsx:41]: I didn't change  the file but I might have run lint it must have changed it

File: src/hooks/useKeyLongPressed.ts (hunk 3 above)
Comment (bhtibrewal) [reply src/hooks/useKeyLongPressed.ts:23]: It is just changed because of lint, the only change is single quotes to double quotes

File: src/hooks/useKeyLongPressed.test.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      jest.advanceTimersByTime(300);
+    });
+
+    expect(result.current[0]).toBe('a');
+  });
+
+  it('should clear the keyLongPressed state when the key is released before 300ms', () => {
+    const { result } = renderHook(() => useKeyLongPressed());
+
+    act(() => {
```
[the remark this answers is not in the document]
Comment (bhtibrewal) [reply src/hooks/useKeyLongPressed.test.tsx:27]: I have a habit of keeping the file and test file alongside so that's why I missed it, will move it there
